### PR TITLE
adding support for sentinel config

### PR DESCRIPTION
**Pull request by RiRa12621**
Support for redis sentinel configuration as documented on https://sensuapp.org/docs/latest/redis used for HA clustering and failover

### CODE REVIEW

**Comment by RiRa12621** [comment]
[Build Status](https://travis-ci.org/RiRa12621/sensu-puppet)
build passing after fixing the typo

File: .travis.yml [hunk 1]
```diff
@@ -27,4 +27,4 @@ matrix:
     env: PUPPET_GEM_VERSION="~> 4"
 notifications:
   email:
-    - james@lovedthanlost.net
+    - rrackow@ebay.com
```
Comment (jlambert121) [inline .travis.yml]: Would you please revert this change?

**Comment by jlambert121** [comment]
Closes #499 

I'm good to merge this if you'd undo the travis.yaml change and squish everything into a single commit.

**Comment by RiRa12621** [comment]
for the ease of it i'll create a new clean PR with just one commit to it

**Comment by RiRa12621** [comment]
Ok good to go

**Comment by RiRa12621** [comment]
wait a little more please

i figured some problems with it while extensively testing it

**Comment by RiRa12621** [comment]
Ok I debugged it and tested this to the most possible extend

**Comment by RiRa12621** [comment]
let me just make this one commit and then we’re good to go
but please feel free to test beforehand

**Comment by modax** [comment]
Hello,

what is the status of this PR? For me, proper redis HA support is a major new feature of sensu 0.23.2 and it would be great to see this PR completed, merged and maybe even a new version of puppet-sensu released. Thank you!.

**Comment by modax** [comment]
@RiRa12621 I'm no puppet-sensu developer, but your PR seems to have some unrelated (unless I'm mistaken) default value changes in manifests/init.pp and manifests/check.pp.

**Comment by RiRa12621** [comment]
@modax there’s a “bug” or known issue, that when you have IPv6 enabled on the host and have redis running locally, it does resolve to 127.0.0.1 when you put ‘localhost’ in the config and thus connection to redis fails, so i changed that.

**Comment by modax** [comment]
@RiRa12621 yeah, localhost -> 127.0.0.1 is ok, I think. It is already in current sensu-puppet master as far as I can tell (try rebasing). But as well there are like:

```diff
diff --git a/manifests/check.pp b/manifests/check.pp
index bbff46e..14a4fd5 100644
--- a/manifests/check.pp
+++ b/manifests/check.pp
@@ -85,7 +85,7 @@ define sensu::check(
   $ensure              = 'present',
   $type                = undef,
   $handlers            = undef,
-  $standalone          = true,
+  $standalone          = false,
   $interval            = 60,
   $occurrences         = undef,
   $refresh             = undef,
```
```
diff --git a/manifests/init.pp b/manifests/init.pp
index 8794b5e..27eaa40 100644
--- a/manifests/init.pp
+++ b/manifests/init.pp
@@ -369,13 +369,13 @@ class sensu (
   $redis_host                     = '127.0.0.1',
   $redis_port                     = 6379,
   $redis_password                 = undef,
-  $redis_reconnect_on_error       = false,
+  $redis_reconnect_on_error       = true,
   $redis_db                       = 0,
   $redis_auto_reconnect           = true,
   $redis_sentinels                = [],
```

While redis_reconnect_on_error  should default to true (since sensu 0.18), probably it should not be hidden in the same commit.

**Comment by RiRa12621** [comment]
tbh i cannot remember why i made reconnect on error default true but for standalone: this broke around 1,5k hosts for me so i changed it back to false
but valid point that it possibly shouldn’t be in there

File: manifests/check.pp [hunk 2]
```diff
@@ -85,7 +85,7 @@
   $ensure              = 'present',
   $type                = undef,
   $handlers            = undef,
-  $standalone          = true,
+  $standalone          = false,
```
Comment (jaxxstorm) [inline manifests/check.pp]: Why is this changed?

**Comment by modax** [comment]
@jaxxstorm: I suggest you looking at https://github.com/sensu/sensu-puppet/pull/509 PR instead. This PR has a couple of unrelated changes (see @RiRa12621 explanation of a few of them at https://github.com/sensu/sensu-puppet/pull/496#issuecomment-221870963). Functionally wise my PR #509 provides everything this one provides + more. I think @RiRa12621 agrees.

**Comment by jaxxstorm** [comment]
Agreed, closing this in favour of #509

**Comment by RiRa12621** [comment]
@modax is fully right
